src/Features/Core/Portable/AddImport/PackageSourceHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Packaging;
+
+namespace Microsoft.CodeAnalysis.AddImport
+{
+    internal static class PackageSourceHelper
+    {
+        private const string NugetOrg = "nuget.org";
+        public const string NugetOrgSourceName = "::nuget::";
+
+        public static IEnumerable<(string sourceName, string sourceUrl)> GetPackageSources(ImmutableArray<PackageSource> packageSources)
+        {
+            // Package source names are user configurable, but various operations and background tasks process
+            // only the nuget source, so we ignore the user defined name for nuget.org so we can identify it later.
+
+            var foundNugetOrg = false;
+            foreach (var packageSource in packageSources)
+            {
+                // If the user has multiple sources from nuget.org, we only need one of them to be special
+                if (!foundNugetOrg)
+                {
+                    var host = new Uri(packageSource.Source).Host;
+                    if (host.Equals(NugetOrg, StringComparison.OrdinalIgnoreCase) ||
+                        host.EndsWith($".{NugetOrg}", StringComparison.OrdinalIgnoreCase))
```

Comment:
what's an example of this?

Reply:
This is for the standard nuget.org feed url, for which `api.nuget.org` is the host.

Reply:
Doc me Amadeus